dex/networks/erc20/contracts/ERC20SwapV0.sol

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: BlueOak-1.0.0
+// pragma should be as specific as possible to allow easier validation.
+pragma solidity = 0.8.6;
+
+interface ERC20 {
+    function transfer(address recipient, uint256 amount) external returns (bool);
+    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
```

Comment:
USDT's ERC20 is not quite an ERC20 since it doesn't have the returns from these functions.  https://etherscan.io/address/0xdac17f958d2ee523a2206206994597c13d831ec7#code  Will that he a challenge to accomodate with this `ERC20Swap` contract?

Reply:
May be a good reference for other oddities: https://github.com/d-xo/weird-erc20

Reply:
I don't think that will have an effect since the return type of a function is not part of its signature, but I'll test it out.

Reply:
With my previous implementation, `transfer` and `transferFrom` actually don't work when the functions don't return anything. The current implementation fixes it.

I'm not sure if there's a general solution to checking a blocklist.. that's something we could only implement individually for each token.

Reply:
As `token` in the contract would correspond to a known contract, we could make a list/map of blacklists later and apply if the address matches.

Reply:
Would relying on the `token.call` to fail on blacklisted addresses work? I would think a blacklisted address is prevented from sending or receiving funds?